src/main/java/core/basesyntax/SalaryInfo.java

```diff
@@ -1,7 +1,36 @@
 package core.basesyntax;
 
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+
 public class SalaryInfo {
     public String getSalaryInfo(String[] names, String[] data, String dateFrom, String dateTo) {
-        return null;
+        StringBuilder result = new StringBuilder("Report for period " + dateFrom + " - " + dateTo);
+        int[] salary = new int[names.length];
+        DateTimeFormatter formatter = DateTimeFormatter.ofPattern("dd.MM.yyyy");
+
+        for (String datum : data) {
+            String[] str = datum.split(" ");
+            String name = str[1];
+            LocalDate date = LocalDate.parse(str[0], formatter);
+            LocalDate localDateFrom = LocalDate.parse(dateFrom, formatter);
+            LocalDate localDateTo = LocalDate.parse(dateTo, formatter);
+            int salaryInDay = (Integer.parseInt(str[2]) * Integer.parseInt(str[3]));
+            if ((date.isAfter(localDateFrom) || date.isEqual(localDateFrom))
+                    && (date.isBefore(localDateTo) || date.isEqual(localDateTo))) {
+                for (int j = 0; j < names.length; j++) {
+                    if (names[j].equals(name)) {
+                        salary[j] += salaryInDay;
+                    }
+                }
+            }
+        }
+        for (int i = 0; i < names.length; i++) {
```

Comment:
Too many [loops](https://mate-academy.github.io/jv-program-common-mistakes/java-core/exceptions/salary-info#dont-create-redundant-loops-two-is-enough)